Add feature from ZX branch

ikhoeyZX opened the pull request:
- Recenter overlay window and button
- Add close button in overlay menu (because [x] too small and sometimes hard to pressed)
- Allow scrolling in controller menu (because i did tried slide scroll slider at right, sometimes work and sometimes undetected because too small)
- hide "debug" in menubar (it useless for end user and configuration button cant be pressed when multiwindows or freewindow mode because freewindows/multiwindow overide it or camera block it) > this one usefull for people have issue with builtin driver (adreno user) because they cant run fullscreen before install driver
- allow resize joystick only instead all button
- allow fallback to default driver in case custom driver broken (adreno only)
- allow performance mode (may fix some device that have issue with locked in small core than big core)
- cutout mode set to "shortEdges" (so it will draw full screen now, no black bar)
- fix shortcut icon issue
- allow ziped sdl2 files to reduce upload size (from ~24mb to 3.8mb)
- add usb  controller support for A14
- fix texture issue (vulkan) in non bcn supported devices
- add invert gyro and enable/disable motion sensors

example preview:
![Screenshot_20241112_022459_Vita3K](https://github.com/user-attachments/assets/5fedddd3-85a8-41b3-aa83-18d89d3abeac)

https://github.com/user-attachments/assets/4c855d95-75c9-4cf1-92db-df9d691dc4ea

![IMG-20240731-WA0000](https://github.com/user-attachments/assets/1a0da624-8df1-449b-a699-017af602f09b)
![IMG-20240731-WA0001](https://github.com/user-attachments/assets/19916ac4-c19a-4118-94c2-c426230efd75)
![IMG-20240731-WA0003](https://github.com/user-attachments/assets/c34e7092-f625-433f-b4b9-3c83a9085956)

https://github.com/user-attachments/assets/5181939e-0594-4760-bd72-e76942a9beb6

ikhoeyZX committed:
Update config.h

ikhoeyZX committed:
Update about_dialog.cpp

ikhoeyZX committed:
Update controllers_dialog.cpp

ikhoeyZX committed:
Update controls_dialog.cpp

ikhoeyZX committed:
Update main_menubar.cpp

ikhoeyZX committed:
Update state.h

ikhoeyZX committed:
Update settings_dialog.cpp

ikhoeyZX committed:
Update live_area.cpp

Increase button size because it's too small in touchscreen devices (and some small display hard to press it)

ikhoeyZX committed:
Fallback adreno driver

Use built-in driver instead custom driver when custom driver not supported/broken so no need remove config.yml

ikhoeyZX committed:
Update Emulator.java

joystick scale

ikhoeyZX committed:
Update InputOverlay.java

split joystick scale and global scale

ikhoeyZX committed:
Update config.h

ikhoeyZX committed:
Update gui.cpp

ikhoeyZX committed:
Update controls_dialog.cpp

ikhoeyZX committed:
Update functions.h

ikhoeyZX committed:
Update AndroidManifest.xml

ikhoeyZX committed:
Create game_mode_config.xml

ikhoeyZX committed:
Update styles.xml

Cutout fix

Zangetsu38 commented on vita3k/gui/src/live_area.cpp:
```diff
@@ -1062,7 +1062,7 @@ void draw_live_area_screen(GuiState &gui, EmuEnvState &emuenv) {
     const auto SELECT_POS = ImVec2(GATE_POS.x + (5.f * SCALE.y), GATE_POS.y + (2.f * SCALE.y));
 
-    const auto BUTTON_SIZE = ImVec2(72.f * SCALE.x, 30.f * SCALE.y);
+    const auto BUTTON_SIZE = ImVec2(80.f * SCALE.x, 55.f * SCALE.y);
```
no need change it, problem here is not size, is how button size is scalled on android, because code for respect aspect ratio is disable on android
i not have this problem in my side
Buton size is all time same and fine

ikhoeyZX replied:
in my phone that button was hard to press, most often vita3k notification got pressed instead esc.
or should i move a bit lower instead resize it?

ikhoeyZX committed:
Update Emulator.java

get path from pref-path for icons

ikhoeyZX committed:
Update app_context_menu.cpp

ikhoeyZX committed:
use zip package for sdl2

distribute in libSDL2.zip instead unpack all files
because unpacked libSDL2.a too big for mobile data plan (12mb uncompressed and 2mb compressed)

ikhoeyZX committed:
Update texture.cpp

i did test this and seems like correctly set conversion base format increase fps in some games (in mali gpu)

ikhoeyZX committed:
Update SDL.java

ikhoeyZX committed:
Update SDLActivity.java

ikhoeyZX committed:
Update SDLControllerManager.java

ikhoeyZX committed:
Update SDLActivity.java

ikhoeyZX committed:
Update HIDDeviceManager.java

ikhoeyZX committed:
Update texture.cpp

color should be same like adreno phones
some games fixed motion blur effect

ikhoeyZX committed:
Update texture.cpp

run gamma correction first then convert to linear of not support it

ikhoeyZX committed:
Update initial_setup.cpp

fix: warning: ISO C++11 does not allow conversion from string literal to 'char *const' [-Wwritable-strings]

ikhoeyZX committed:
Update controls_dialog.cpp

add menu invert gyro and disable motion sensor

ikhoeyZX committed:
Update controllers_dialog.cpp

add enable/disable motion and invert gyro

ikhoeyZX committed:
Update ctrl.cpp

ikhoeyZX committed:
Update SceMotion.cpp

ikhoeyZX committed:
Update config.h

add config

ikhoeyZX committed:
Update config.h

ikhoeyZX committed:
Update state.h

ikhoeyZX committed:
Update private.h

ikhoeyZX committed:
Update gui.cpp

ikhoeyZX committed:
Update state.h

ikhoeyZX committed:
Update state.h

fix wrong placement

ikhoeyZX committed:
Update controls_dialog.cpp

ikhoeyZX committed:
Update controls_dialog.cpp

ikhoeyZX committed:
Update SceThreadmgr.cpp

rebase https://github.com/Vita3K/Vita3K/pull/3397/commits/ffa8ae0775ce636b457141f488e932a40edf86f3

ikhoeyZX committed:
Update SceThreadmgr.h

ikhoeyZX committed:
Update SceLibKernel.cpp

ikhoeyZX committed:
Update controls_dialog.cpp

ikhoeyZX committed:
Update types.h

ikhoeyZX committed:
Update thread.cpp

ikhoeyZX committed:
Update thread.cpp

levelofmy commented:
能多添加新的着色器？

ikhoeyZX committed:
Merge branch 'Vita3K:android' into GUI_little_fix

ikhoeyZX committed:
Update state.h

ikhoeyZX committed:
Update pipeline_cache.cpp

ikhoeyZX committed:
Update surface_cache.cpp

ikhoeyZX committed:
Update creation.cpp

ikhoeyZX committed:
Update screen_renderer.cpp

ikhoeyZX committed:
Update build.gradle

ikhoeyZX committed:
Update android.yml
